<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Download</h1>
</div>

This page contains downloadable data for "Multiomics dissection of healthy human aging" experiment.
<div>
    <ul>
        <li><a href="#download-chipseq">ChIP-seq</a></li>
        <li><a href="#download-methylation">DNA Methylation</a></li>
        <li><a href="#download-rnaseq">RNA-Seq</a></li>
        <li><a href="#download-metabolomics">Metabolomics</a></li>
        <li><a href="#download-proteomics">Proteomics</a></li>
        <li><a href="#download-public-data">Public Data</a></li>
    </ul>
</div>

<br>
<h2 id="download-chipseq">ChIP-Seq</h2>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th>Alignment</th>
        <th>QC</th>
        <th>Bigwigs</th>
        <th>Peaks</th>
        <th>Visual Labels</th>
        <th>Peak Caller Models</th>
    </tr>
    </thead>
    <tbody>
    @TABLE_CHIPSEQ@
    </tbody>
</table>
Files included in this table:
<ul>
    <li>
        Alignment - aligned reads in BED.GZ format,
        <br>aligned without multimappers on <a
            href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=hg19">hg19</a> (GRCh37 Genome Reference Consortium Human
        Reference 37).
        <br>
        Raw reads in FASTQ format will be available upon request at <a href="https://www.ncbi.nlm.nih.gov/gap">dbGAP</a>.

    </li>
    <li>
        QC - library QC information by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a>.
    </li>
    <li>
        BigWigs - visualization of reads for each donor in BigWig format.
    </li>
    <li>
        Peaks - BED files with peaks obtained using novel semi-supervised peak calling approach.
    </li>
    <li>
        Visual Labels - manual annotations of genome locations as peaks, valleys or peak shores,
        used to obtain optimal peaks for a given sample.
    </li>
    <li>
        Peak Caller Models - trained <a href="span.html">SPAN</a> Peak Analyzer models for each sample.
    </li>
</ul>
<p>
    Learn more about <a href="tools.html">visual peak calling</a> and find out how to apply this approach to your data
    in How-To <a href="howto.html">tutorial</a>.
</p>

<br>
<h2 id="download-methylation">DNA Methylation</h2>
<table class="table table-bordered table-nonfluid">
    <tbody>
    <tr>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/per_donor">Alignment</a>
        </th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/qc">QC</a></th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/filtered_cytosines.bed">Filtered
            Cytosines</a></th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/dmrs">Methylation Level</a>
        </th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/dmrs">Coverage</a></th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/methylation/TableSX_annotated_dmrs.xlsx">DMRs</a>
        </th>
    </tr>
    </tbody>
</table>

Files included in this table:
<ul>
    <li>
        Alignment - Reduced Representation Bisulfite Sequencing data, <br>
        aligned on <a
            href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=hg19">hg19</a> (GRCh37 Genome Reference Consortium Human
        Reference 37).<br>
    </li>
    <li>
        QC - library QC information by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a>.
    </li>
    <li>Filtered Cytosines - filtered cytosines in BED format, used in downstream analysis.
        <br>Filtration criteria: coverage &geq; 3, absolute difference in methylation level &geq; 0.025.
    </li>
    <li>
        Methylation Level - methylation level for each cytosine after filtration.
    </li>
    <li>
        Coverage - coverage for each cytosine after filtration.
    </li>
    <li>
        DMRs - Differentially Methylated Regions provided by <a
            href="http://smithlabresearch.org/manuals/methpipe-manual.pdf">MethPipe</a> methylation analysis pipeline.
    </li>
</ul>
<br>
<h2 id="download-rnaseq">RNA-Seq</h2>
<table class="table table-bordered table-nonfluid">
    <tbody>
    <tr>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/bed">Alignment</a></th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/fastqc">QC</a></th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/bigwig">BigWigs</a></th>
        <th><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/counts">Counts</a></th>
        <th>
            <a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/aging_10vs10.rda.">Phantasus</a>
        </th>
    </tr>
    </tbody>
</table>

Files included in this table:
<ul>
    <li>
        Alignment - RNA-Seq data, <br>
        aligned on <a
            href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=hg19">hg19</a> (GRCh37 Genome Reference Consortium Human
        Reference 37).<br>
    </li>
    <li>
        QC - library QC information by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a>.
    </li>
    <li>
        BigWigs - visualization tracks.
    </li>
    <li>
        Counts - raw counts produced by <a
            href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> package, used in
        downstream analysis.
    </li>
    <li>
        <code>RDA</code> file, which can be uploaded to <a href="https://artyomovlab.wustl.edu/phantasus/">Phantasus</a>
        RNA-Seq
        online analysis tool.
    </li>
    <li>
        DMRs - Differentially Methylated Regions provided by <a
            href="http://smithlabresearch.org/manuals/methpipe-manual.pdf">MethPipe</a> methylation analysis pipeline.
    </li>
</ul>

<br>
<h2 id="download-metabolomics">Metabolomics</h2>
<p>
    Table with differential metabolomics data analysis is available in <a
        href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/TableS5_metabolite_values_scaled_intensities.xlsx">supplementary
    materials.</a>
</p>
<br>
<h2 id="download-proteomics">Proteomics</h2>
<p>
    Table with differential proteomics data analysis is available in <a
        href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/TableS3_somascan_values_log2RFU.WUS18-007_Artymov.somamer_in_row.xlsx">supplementary
    materials.</a>
</p>
<br>
<hr>
<h2 id="download-public-data">Public Data</h2>
<p>Public data used in "Multiomics dissection of healthy human aging" experiment.</p>

<h4>ENCODE CD14+ ChIP-seq</h4>
CD14+ classical monocytes ChIP-seq data available in ENCODE database.
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th>GSM</th>
        <th>Bigwigs</th>
        <th>Peaks</th>
        <th>Visual Labels</th>
    </tr>
    </thead>
    <tbody>
    @TABLE_ENCODE@
    </tbody>
</table>

<p>Files included in this table:</p>

<ul>
    <li>GSM - GEO accession id for each sample.</li>
    <li>
        BigWigs - visualization of reads for each donor in BigWig format.
    </li>
    <li>
        Peaks - BED files with peaks obtained using novel semi-supervised peak calling <a href="tools.html">approach</a>.
    </li>
    <li>
        Visual Labels - manual annotations of genome locations as peaks, valleys or peak shores,
        used to obtain optimal peaks for a given sample.
    </li>
</ul>

<h4>ChromHMM</h4>
<p>
    18-states <a href="http://compbio.mit.edu/ChromHMM/">ChromHMM</a> model markup for CD14 monocytes can be downloaded
    <a href="https://www.encodeproject.org/annotations/ENCSR907LCD">from ENCODE project site</a>.
</p>

<h4>Blueprint DNA-Methylation</h4>
<p>
    <a href="http://dcc.blueprint-epigenome.eu/#/home">BLUEPRINT Epigenome consortium data</a>
    for monocytes DNA methylation.
</p>
