<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Download</h1>
</div>

This page contains downloadable data for the experiment.
<div>
    <ul>
        <li><a href="#download-chipseq">ChIP-seq</a></li>
        <li><a href="#download-methylation">DNA Methylation</a></li>
        <li><a href="#download-rnaseq">RNA-Seq</a></li>
        <li><a href="#download-metabolomics">Metabolomics</a></li>
        <li><a href="#download-proteomics">Proteomics</a></li>
        <li><a href="#download-public-data">Public Data</a></li>
    </ul>
</div>

<br>
<h2 id="download-chipseq">ChIP-Seq</h2>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th>Alignment</th>
        <th>QC</th>
        <th>BigWigs</th>
        <th>Peaks</th>
        <th>Visual Labels</th>
        <th>Peak Caller Models</th>
    </tr>
    </thead>
    <tbody>
    @TABLE_CHIPSEQ@
    </tbody>
</table>
Files included in this table:
<ul>
    <li>
        Alignment - aligned reads in BED.GZ format,
        <br>aligned without multimappers on <a
            href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=hg19">hg19</a> (GRCh37 Genome Reference Consortium Human
        Reference 37).
        <br>
        Raw reads in FASTQ format will be available upon request at <a href="https://www.ncbi.nlm.nih.gov/gap">dbGAP</a>.

    </li>
    <li>
        QC - library QC information by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a>.
    </li>
    <li>
        BigWigs - visualization of reads for each donor in bigWig format.
    </li>
    <li>
        Peaks - BED files with peaks obtained using a novel semi-supervised peak calling approach.
    </li>
    <li>
        Visual Labels - manual annotations of genome locations as peaks, valleys or peak shores,
        used to obtain optimal peaks for a given sample.
    </li>
    <li>
        Peak Caller Models - trained <a href="https://research.jetbrains.org/groups/biolabs/tools/span-peak-analyzer"
                                        target="_blank">SPAN</a> Peak Analyzer models for each sample.
    </li>
</ul>
<p>
    Learn more about <a href="tools.html">visual peak calling</a> and find out how to apply this approach the in How-To
    <a href="howto.html">tutorial</a>.
</p>

<br>
<h2 id="download-methylation">DNA Methylation</h2>
<table class="table table-bordered table-nonfluid">
    <tbody>
    <tr>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/per_donor">Methylation
            Calls</a>
        </td>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/qc">QC</a></td>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/filtered_cytosines.bed">Filtered
            Cytosines</a></td>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/filtered_cytosines_freq.tsv">Methylation
            Level</a>
        </td>
        <td>
            <a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/filtered_cytosines_coverage.tsv">Coverage</a>
        </td>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rrbs/TableS9_annotated_dmrs.xlsx">DMRs</a>
        </td>
    </tr>
    </tbody>
</table>

Files included in this table:
<ul>
    <li>
        Methylation Calls - preprocessed Reduced Representation Bisulfite Sequencing data, <br>
        aligned on <a
            href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=hg19">hg19</a> (GRCh37 Genome Reference Consortium Human
        Reference 37).<br>
    </li>
    <li>
        QC - summarised RRBS processing pipeline QC reports provided for each sample by the facility.
    </li>
    <li>Filtered Cytosines - filtered cytosines in BED format, used in downstream analysis.
        <br>Filtration criteria: mean coverage across samples &geq; 10.
    </li>
    <li>
        Methylation Level - table with per donor methylation levels for remained after filtration cytosines.
    </li>
    <li>
        Coverage - table with per donor coverage for remained after filtration cytosines.
    </li>
    <li>
        DMRs - annotated Differentially Methylated Regions generated by the <a
            href="https://smithlabresearch.org/manuals/methpipe-manual.pdf">MethPipe</a> pipeline.
    </li>
</ul>
<br>
<h2 id="download-rnaseq">RNA-Seq</h2>
<table class="table table-bordered table-nonfluid">
    <tbody>
    <tr>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/bed">Alignment</a></td>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/fastqc">QC</a></td>
        <td><a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/counts">Counts</a></td>
        <td>
            <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_10vs10">Phantasus</a>
        </td>
    </tr>
    </tbody>
</table>

Files included in this table:
<ul>
    <li>
        Alignment - RNA-Seq data, <br>
        aligned on <a
            href="https://genome.ucsc.edu/cgi-bin/hgGateway?db=hg19">hg19</a> (GRCh37 Genome Reference Consortium Human
        Reference 37).<br>
        Raw reads in FASTQ format will be available upon request at <a href="https://www.ncbi.nlm.nih.gov/gap">dbGAP</a>.
    </li>
    <li>
        QC - library QC information by <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a>.
    </li>
    <li>
        Counts - raw counts produced by <a href="https://htseq.readthedocs.io/en/release_0.10.0/">HTSeq</a>, <a
            href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> normalised counts, and
        log-transformed quantile normalised counts used in downstream analysis.
    </li>
    <li>
        Phantasus - <code>RDA</code> file, which can be uploaded to <a
            href="https://artyomovlab.wustl.edu/phantasus/phantasus-tutorial.html">Phantasus</a>
        RNA-Seq online analysis tool. Explore online <a
            href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_10vs10">here</a>.
    </li>
</ul>

<br>
<h2 id="download-metabolomics">Metabolomics</h2>
<p>
    Table with metabolomics data and differential analysis results is available in <a
        href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/TableS5_metabolite_values_scaled_intensities.xlsx">supplementary
    materials.</a>
</p>
<br>
<h2 id="download-proteomics">Proteomics</h2>
<p>
    Raw microarray data is available in <a
        href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/Somalogic_generated_adat_WUS18_007/">supplementary
    materials.</a>
</p>
<br>
<hr>
<h2 id="download-public-data">Public Data</h2>
<p>Public data used in "Multiomics dissection of healthy human aging" experiment.</p>

<h4>ENCODE CD14+ ChIP-seq</h4>
CD14+ classical monocytes ChIP-seq data available in the ENCODE database.
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th>GSM</th>
        <th>BigWigs</th>
        <th>Peaks</th>
        <th>Visual Labels</th>
    </tr>
    </thead>
    <tbody>
    @TABLE_ENCODE@
    </tbody>
</table>

<p>Files included in this table:</p>

<ul>
    <li>GSM - GEO accession id for each sample.</li>
    <li>
        BigWigs - visualization of reads for each donor in bigWig format.
    </li>
    <li>
        Peaks - BED files with peaks obtained using a novel semi-supervised peak calling <a
            href="tools.html">approach</a>.
    </li>
    <li>
        Visual Labels - manual annotations of genome locations as peaks, valleys, or peak shores,<br>
        used to obtain optimal peaks for a given sample.
    </li>
</ul>

<h4>ChromHMM</h4>
<p>
    18-states <a href="http://compbio.mit.edu/ChromHMM/">ChromHMM</a> model markup for CD14 monocytes can be downloaded
    from
    the <a href="https://www.encodeproject.org/annotations/ENCSR907LCD">ENCODE</a> project site. <br>
    Here is <a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/ENCFF072SCA.bigBed">the
    file</a> used in our study.
</p>


<h4>BLUEPRINT DNA-Methylation Data</h4>
<p>
    This section contains BLUEPRINT Epigenome consortium methylation data for CD14-positive, CD16-negative classical
    monocytes. <br>
    These files are accessible at BLUEPRINT <a href="http://dcc.blueprint-epigenome.eu/#/files">data portal</a> by
    applying filters:<br>
    <code>Experiment=Bisulfite-seq</code>, <code>Type=Methylation signal</code>, and <code>Cell type=CD14-positive
    CD16-negative classical monocyte</code>.
    <br>
    Bisulfite Sequencing protocol description is available <a
        href="http://dcc.blueprint-epigenome.eu/#/md/bs_seq_grch38">here</a>.
</p>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Download</th>
        <th>Source</th>
        <th>Name</th>
        <th>Sex</th>
        <th>Metadata</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <td>
            <a href="http://ftp.ebi.ac.uk/pub/databases/blueprint/data/homo_sapiens/GRCh38/cord_blood/C005PS/CD14-positive_CD16-negative_classical_monocyte/Bisulfite-Seq/CNAG/C005PS51.CPG_methylation_calls.bs_call.GRCh38.20160531.bw">BigWig</a>
        </td>
        <td class="ng-binding">cord blood</td>
        <td class="ng-binding">C005PS</td>
        <td class="ng-binding">Female</td>
        <td><a href="http://dcc.blueprint-epigenome.eu/#/experiments/ERX242612">ERX242612</a></td>
    </tr>
    <tr>
        <td>
            <a href="http://ftp.ebi.ac.uk/pub/databases/blueprint/data/homo_sapiens/GRCh38/cord_blood/S000RD/CD14-positive_CD16-negative_classical_monocyte/Bisulfite-Seq/CNAG/S000RD54.CPG_methylation_calls.bs_call.GRCh38.20160531.bw">BigWig</a>
        </td>
        <td class="ng-binding">cord blood</td>
        <td class="ng-binding">S000RD</td>
        <td class="ng-binding">Male</td>
        <td><a href="http://dcc.blueprint-epigenome.eu/#/experiments/ERX242609">ERX242609</a></td>
    </tr>
    <tr>
        <td>
            <a href="http://ftp.ebi.ac.uk/pub/databases/blueprint/data/homo_sapiens/GRCh38/venous_blood/C000S5/CD14-positive_CD16-negative_classical_monocyte/Bisulfite-Seq/CNAG/C000S5A1bs.CPG_methylation_calls.bs_call.GRCh38.20160531.bw">BigWig</a>
        </td>
        <td class="ng-binding">venous blood</td>
        <td class="ng-binding">C000S5</td>
        <td class="ng-binding">Male</td>
        <td><a href="http://dcc.blueprint-epigenome.eu/#/experiments/ERX204011">ERX204011</a></td>
    </tr>
    <tr>
        <td>
            <a href="http://ftp.ebi.ac.uk/pub/databases/blueprint/data/homo_sapiens/GRCh38/venous_blood/C0010K/CD14-positive_CD16-negative_classical_monocyte/Bisulfite-Seq/CNAG/C0010KA2bs.CPG_methylation_calls.bs_call.GRCh38.20160531.bw">BigWig</a>
        </td>
        <td class="ng-binding">venous blood</td>
        <td class="ng-binding">C0010K</td>
        <td class="ng-binding">Female</td>
        <td><a href="http://dcc.blueprint-epigenome.eu/#/experiments/ERX204012">ERX204012</a></td>
    </tr>
    <tr>
        <td>
            <a href="http://ftp.ebi.ac.uk/pub/databases/blueprint/data/homo_sapiens/GRCh38/venous_blood/C001UY/CD14-positive_CD16-negative_classical_monocyte/Bisulfite-Seq/CNAG/C001UYA3bs.CPG_methylation_calls.bs_call.GRCh38.20160531.bw">BigWig</a>
        </td>
        <td class="ng-binding">venous blood</td>
        <td class="ng-binding">C001UY</td>
        <td class="ng-binding">Male</td>
        <td><a href="http://dcc.blueprint-epigenome.eu/#/experiments/ERX204014">ERX204014</a></td>
    </tr>
    <tr>
        <td>
            <a href="http://ftp.ebi.ac.uk/pub/databases/blueprint/data/homo_sapiens/GRCh38/venous_blood/C004SQ/CD14-positive_CD16-negative_classical_monocyte/Bisulfite-Seq/CNAG/C004SQ51.CPG_methylation_calls.bs_call.GRCh38.20160531.bw">BigWig</a>
        </td>
        <td class="ng-binding">venous blood</td>
        <td class="ng-binding">C004SQ</td>
        <td class="ng-binding">Female</td>
        <td><a href="http://dcc.blueprint-epigenome.eu/#/experiments/ERX242611">ERX242611</a></td>
    </tr>
    </tbody>
</table>
<p>Materials included in this table:</p>

<ul>
    <li>
        BigWig - visualization of methylation level in bigWig format.
    </li>
    <li>
        Source - tissue type, material used in the experiment.
    </li>
    <li>
        Name - subject id.
    </li>
    <li>
        Sex - subject sex.
    </li>
    <li>
        Metadata - experiment summary.
    </li>
</ul>

<h4>Expression Data</h4>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>GSM</th>
        <th>Dataset</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <th><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE32719">GSE32719</a></th>
        <td>Expression data from human bone marrow hematopoietic stem cells</td>
    </tr>
    <tr>
        <th><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE87105">GSE87105</a></th>
        <td>Expression data from human skeletal muscle cells</td>
    </tr>
    </tbody>
</table>
<p>Explore these datasets in Phantasus <a
        href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_public_datasets">here</a>.</p>
