<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Explore ChIP-Seq Data</h1>
</div>

This page contains materials for all histone modifications processed in "Multiomics dissection of healthy human aging" experiment.

<h2>Explore</h2>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th>UCSC Genome Browser</th>
        <th>Washu Epigenome Browser</th>
    </tr>
    </thead>
    <tbody>
    @TABLE@
    </tbody>
</table>
Materials included in this table:
<ul>
    <li>
        Session Type - Basic or Extended.<br>
        Basic session contains reads visualization in BigWig format, individual peaks
        for each sample, and labels used in <a href="tools.html">semi-supervised peaks analysis</a>.
        Also it contains ENCODE tracks for CD14+ monocytes.
        <br>
        Extended session contains additional peaks produced by golden standard peak callers, i.e. <a
            href="https://github.com/taoliu/MACS">MACS2</a>
        and <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152844/">SICER</a>.
    </li>
    <li>
        Preconfigured sessions for <a href="https://genome.ucsc.edu">UCSC Genome Browser</a>.
    </li>
    <li>
        Preconfigured sessions for <a
            href="https://epigenomegateway.wustl.edu">WashU Epigenome Browser</a>.
    </li>
</ul>


<h2>Sessions</h2>
Along with preconfigured online genome browser sessions, we prepared sessions for desktop genome browsers:
<a href="https://software.broadinstitute.org/software/igv/">IGV Genome Browser</a> and
<a href="tools.html">JBR Genome Browser</a>. Click on the corresponding link to download application.
<br/>
NOTE: <b>JBR genome browser</b> can be used as integrated solution for <a href="tools.html">peak calling</a>.

<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th>IGV/JBR Session</th>
        <th>UCSC Custom Tracks</th>
    </tr>
    </thead>
    <tbody>
    @TABLE2@
    </tbody>
</table>
Materials included in this table:
<ul>
    <li>
        Downloadable custom tracks file to create custom session for UCSC genome browser.
    </li>
    <li>
        Downloadable session, which can opened in both IGV and JBR genome browsers.
    </li>
</ul>

<h2>Download</h2>
All the data used in sessions is available in <a href="download_chipseq.html">Download ChIP-Seq</a> and <a
        href="public_data.html">Download Public Data </a> sections.




