<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Explore</h1>
</div>

This page contains materials for the data processed.
<div>
    <ul>
        <li><a href="#explore-chipseq">ChIP-seq</a></li>
        <li><a href="#explore-rnaseq">RNA-Seq</a></li>
    </ul>
</div>
<h2 id="explore-chipseq">ChIP-Seq</h2>
<ul>
    <li><a href="#view">View</a> data in UCSC genome browser.</li>
    <li>Download preconfigured <a href="#download">session files.</a></li>
    <li>Download <a href="#data">data</a> used in sessions.</li>
</ul>
<h3 id="view">View</h3>
<p>
    Explore preconfigured online sessions in UCSC genome browser in one click.
</p>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th class="text-center">Basic</th>
        <th class="text-center">Extended</th>
    </tr>
    </thead>
    <tbody>
    @TABLE@
    </tbody>
</table>
Materials included in this table:
<ul>
    <li>
        Preconfigured sessions for <a href="https://genome.ucsc.edu">UCSC Genome Browser</a>.
    </li>
    <li>
        Session Type - Basic or Extended.<br>
        <p>
            Basic session contains reads visualization in bigWig format, individual peaks for each sample,
            <br>
            and labels used in <a href="tools.html">semi-supervised peaks analysis</a>.
            <br>
            ENCODE tracks for classical CD14+ monocytes are included for control.
            <br>
            Extended session contains additional peaks produced by golden standard peak callers, i.e. <a
                href="https://github.com/taoliu/MACS">MACS2</a>
            and <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4152844/">SICER</a>.
        </p>
    </li>
</ul>


<h3 id="download">Download</h3>
<p>
    For those who prefer desktop genome browser applications, we prepared dedicated session files for the following
    genome browsers:
    <br>
    <a href="https://software.broadinstitute.org/software/igv/">IGV Genome Browser</a> and
    <a href="https://research.jetbrains.org/groups/biolabs/tools/jbr-genome-browser" target="_blank">JBR Genome Browser</a>.
    <br>
    Click on the corresponding link to download application.
</p>
<p>
    See the section <a href="tools.html">Tools</a> to learn how <code>JBR</code> Genome Browser can be used for
    visual peak calling.
</p>
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Modification</th>
        <th class="text-center">Basic</th>
        <th class="text-center">Extended</th>
    </tr>
    </thead>
    <tbody>
    @TABLE2@
    </tbody>
</table>
Materials included in this table:
<ul>
    <li>
        Preconfigured session file for <code>IGV</code> or <code>JBR</code> genome browsers.
    </li>
    <li>
        Session types and content is described in the <a href="#explore">Explore</a> section.
    </li>
</ul>

<h2 id="explore-rnaseq">RNA-Seq</h2>
<p>
    <a href="https://artyomovlab.wustl.edu/phantasus/?preloaded=aging_10vs10">Explore</a> RNA-seq data in <a href="https://artyomovlab.wustl.edu/phantasus/phantasus-tutorial.html">Phantasus</a>
    - web-application for visual and interactive gene expression analysis.<br>
    Download preconfigured <code>RDA</code> file <a href="https://artyomovlab.wustl.edu/publications/supp_materials/aging/rnaseq/aging_10vs10.rda">here</a>.
</p>

<h2 id="data">Data</h2>
All the data is available in <a href="download_data.html">Download Data</a> section.


