<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Visual Peak Calling How-To</h1>
</div>
<p>Visual peak calling is a semi-supervised algorithm. At first we need to train SPAN models
    for each track locally or using computational cluster. Then visualize Chip-seq signal
    tracks in JBR Genome Browser, load trained SPAN models,
    create annotations and run peak calling tuning procedure.
</p>
<p>
    This tutorial demonstrates how to perform all steps of the visual peak calling algorithm. As
    an example let's take 3 ULI Chip-Seq tracks from
    <a href="index.html">Multiomics dissection of healthy human aging</a> experiment:  OD7, OD8, OD17.
</p>
<div>
    <h2>Content</h2>
    <ul>
        <li><a href="#jbr-howto-prerequisites">Prerequisites</a></li>
        <li><a href="#jbr-howto-load-data">Load Data Into JBR</a></li>
        <li><a href="#jbr-howto-peak-annotating">Annotating Peaks</a></li>
        <li><a href="#jbr-howto-train">Train SPAN Models</a></li>
        <li><a href="#jbr-howto-peak-calling">Visual Peak Calling</a></li>
    </ul>
</div>

<h2 id="jbr-howto-prerequisites">Prerequisites</h2>
<p>
    Install <a href="span.html#span-installation">SPAN</a> peak caller and
    <a href="jbr.html#jbr-installation">JBR</a> Genome Browser.
</p>
<p>
    This tutorial includes all necessary data files. If you want to use it with your own
    data, you need to prepare ChIP-seq signal tracks as:
    <ul>
    <li>BAM (or BED / BED.gz) files for further processing with SPAN.</li>
    <li>BIGWIG files for visualization in JBR browser.
        <br/>
        You can convert BAM / BED signal tracks to BIGWIG using bedtools, samtools and bedGraphToBigWig utilities.</li>
    </ul>
</p>

<h2 id="jbr-howto-train">Train Models</h2>
<p>
    Model training is a time consuming operation but it is performed only once for each ChIP-Seq track.
    Further peak calling will be very fast ~ about 1 sec for one parameters set.
</p>
<p>
    You may skip model training step and download already trained models for this tutorial:
    <ul>
        <li>
            <a href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/zinbra/mlfnbhmm_zhl_hg19_OD7_k4me1_hg19_input_unique_200.tar">
                <mark>SPAN model file for OD7 track</mark>
            </a>
        </li>
        <li>
            <a href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/zinbra/mlfnbhmm_zhl_hg19_OD8_k4me1_hg19_input_unique_200.tar">
                <mark>SPAN model file for OD8 track</mark>
            </a>
        </li>
        <li>
            <a href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/zinbra/mlfnbhmm_zhl_hg19_OD17_k4me1_hg19_input_unique_200.tar">
                <mark>SPAN model file for OD17 track</mark>
            </a>
        </li>
    </ul>
</p>
<p>In order to train models, download data track for donors OD7, OD8, OD17; control track and hg19 chromosomes size annotation file:</p>
<ul>
    <li>
        <a href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/bedgz/H3K4me1/OD7_k4me1_hg19.bed.gz">
            OD7_k4me1_hg19.bed.gz
        </a>
    </li>
    <li>
        <a href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/bedgz/H3K4me1/OD8_k4me1_hg19.bed.gz">
            OD8_k4me1_hg19.bed.gz
        </a>
    </li>
    <li>
        <a href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/bedgz/H3K4me1/OD17_k4me1_hg19.bed.gz">
            OD17_k4me1_hg19.bed.gz
        </a>
    </li>
    <li>
        <a href="http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/bedgz/input/input.bed.gz">
            input.bed.gz
        </a>
    </li>
    <li>
        <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes">
            hg19.chrom.sizes
        </a>
    </li>
</ul>
<p>
    Run SPAN locally or on your computational cluster. If your run SPAN on multiprocessor machine
    configure parallelism level for better performance :
    <br/>
    <code>java -Xmx1G -jar span.jar analyze -c input.bed.gz -t OD7_k4me1_hg19.bed.gz --chrom.sizes hg19.chrom.sizes  --threads 4</code>
    <br/>
    <code>java -Xmx1G -jar span.jar analyze -c input.bed.gz -t OD8_k4me1_hg19.bed.gz --chrom.sizes hg19.chrom.sizes  --threads 4</code>
    <br/>
    <code>java -Xmx1G -jar span.jar analyze -c input.bed.gz -t OD17_k4me1_hg19.bed.gz --chrom.sizes hg19.chrom.sizes  --threads 4</code>

    <br/>
    <br/>
    The latest lines of SPAN output will contain path to trained model file which we will need later.
    <mark>TODO: show SPAN output example</mark>
</p>

<h2 id="jbr-howto-load-data">Load Data Into JBR Genome Browser</h2>

<p>
    Launch JBR Genome Browser. By default it opens new session for latest used genome. If current
    session ins't <b>hg19</b> then create new hg19 session using 'File | New Session..'.
    <img src="jbr_window.png" style="height: 100%; width: 100%; object-fit: contain"/>
</p>
<p>
    To import track open 'File | Load URL(s)...' action
    <br/>
    <img src="jbr_load_url_action.png" style="height: 50%; width: 50%; object-fit: contain" />

</p>
<p>
    Copy the urls below:
    <br/>
    <code id="code-block">
        http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/bigwigs/H3K4me1/h3k4me1_od7.bw
        <br/>
        http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/bigwigs/H3K4me1/h3k4me1_od8.bw
        <br/>
        http://artyomovlab.wustl.edu/publications/supp_materials/aging/chipseq/Y20O20/bigwigs/H3K4me1/h3k4me1_od17.bw
        <br/>
    </code>
    <br/>
    And paste to the dialog:
    <img src="jbr_load_urls_dialog.png" style="height: 100%; width: 100%; object-fit: contain"/>
</p>

<p>
    Let's zoom track and open location <code>chr1:89,108,000-89,650,000</code>:
    <img src="jbr_loc.png" style="height: 100%; width: 100%; object-fit: contain"/>
</p>

<h2 id="jbr-howto-peak-annotating">Annotating Peaks</h2>
<p>
    <mark>TODO: How to add/remove label, export labels</mark>
    <br/>
    <mark>TODO: Labelling recommendations...</mark>
</p>

<h2 id="jbr-howto-peak-calling">Visual Peak Calling</h2>
<p>
    <mark>TODO: After you've got labels two scenarios are possible: 1) Call peaks in JBR browser 2)
        Call peaks in command line using SPAN.
        Peak calling in JBR allows you to perform peak calling tuning iteratively. E.g. annotate
        peaks, tune peak calling, validate results, if
        you're not satisfied with them, you can remove/add additional annotations and rerun tuning.
        Tuning in cmdline could be convenient if you are sure of your annotations, so you could do peak
        calling
        as part of your pipeline script. In this case rerun SPAN with --labels options.
        Also you can pass --labels while training model, SPAN will tune peak calling arguments and
        call peaks after model training
    </mark>
    <mark>TODO: Suggest to run tune action in JBR. Also describe how to change FDR, GAP and do
        peak calling manually.
    </mark>
    <br/>
    <mark>TODO: Optionally suggest to open our labels and try peak calling tuning to view rescued
        tracks
    </mark>
</p>
 <mark>TODO: Open related SPAN models. User is supposed to download models, open them in JBR browser.
    </mark>

