<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Methods</h1>
</div>

<h2>Protocols</h2>
This section contains description of the protocols used.
<ul>
    <li>
        Monocyte isolation (beads etc) - full
    </li>
    <li>
        ULiChipSeq library preparation - full
    </li>
    <li>
        Metabolomics (short - extracted like this and submitted to blabal..)
    </li>
    <li>
        Proteomic (short)
    </li>
    <li>
        RNA-seq (short - rna was extracted like this... and submitted to blalba)
    </li>
    <li>
        DNA methylation (short)
    </li>
</ul>

<h2>Pipelines</h2>
This section contains description of the computational pipelines used.
<ul>
    <li>
        ChipSeq quality control, processing and analysis
    </li>
    <li>
        RNA-seq quality control, processing and analysis
    </li>
    <li>
        DNA methylation data analysis
    </li>
    <li>
        Metabolomics (? - PCA scripts, DE scripts, pathway enrichment)
    </li>
    <li>
        Proteomics(?)
    </li>
</ul>

<h2>Miscellaneous</h2>
<ul>
    <li>
        Peak calling solution description is <a href="tools.html">here</a>.
    </li>
    <li>
        Learn about visual peak calling in step-by-step <a href="howto.html">how-to</a>.
    </li>
    <li>
        Results of semi-supervised visual peak calling to public datasets is available <a href="#">here</a>.
    </li>
</ul>