<!-- To be included inside template.html -->

<div class="page-header">
    <h1>Methods</h1>
</div>

<h2>Protocols</h2>
This section contains description of the protocols used.
<ul>
    <li>
        Monocyte isolation (beads etc) - full
    </li>
    <li>
        <a href="http://artyomovlab.wustl.edu/aging/protocols/ULI-ChIP-Seq.docx">ULI ChIP-Seq library preparation</a>
    </li>
    <li>
        Metabolomics (short - extracted like this and submitted to blabal..)
    </li>
    <li>
        Proteomic (short)
    </li>
    <li>
        <a href="http://artyomovlab.wustl.edu/aging/protocols/RNA Sequencing.docx">RNA-Seq</a>
    </li>
    <li>
        DNA methylation (short)
    </li>
</ul>

<h2>Pipelines</h2>
This section contains description of the computational pipelines used.
<ul>
    <li>
        ChipSeq quality control, processing and analysis
    </li>
    <li>
        RNA-seq quality control, processing and analysis
    </li>
    <li>
        DNA methylation data analysis
    </li>
    <li>
        Metabolomics (? - PCA scripts, DE scripts, pathway enrichment)
    </li>
    <li>
        Proteomics(?)
    </li>
</ul>

<h2>Miscellaneous</h2>
<ul>
    <li>
        Source code for technical ChIP-Seq processing is available on github: <a href="https://github.com/JetBrains-Research/washu">JetBrains-Research/washu</a>.
    </li>
</ul>