<!-- To be included inside template.html -->

<div class="page-header">
    <h1>SPAN Peak Analyzer</h1>
</div>
<pre id="span-ascii"><code>+----------------------------------+
|SPAN Semi-supervised Peak Analyzer|
+----------------------------|/----+
           ,        ,
      __.-'|'-.__.-'|'-.__
    ='=====|========|====='=
    ~_^~-^~~_~^-^~-~~^_~^~^~^
</code></pre>
<p><code>SPAN</code> is a tool for analyzing ChIP-seq data.</p>

<div>
    <h2>Content</h2>
    <ul>
        <li><a href="#span-features">Features</a></li>
        <li><a href="#span-installation">Installation</a></li>
        <li><a href="#span-usage">Usage of SPAN</a></li>
        <blockquote>
            <ul>
                <li><a href="#span-peak-calling">Peak calling</a></li>
                <li><a href="#span-supervised-peak-calling">Supervised Peak Calling</a></li>
                <li><a href="#span-model-fitting">Model Fitting</a></li>
            </ul>
        </blockquote>
        <li><a href="#span-output-files">Output Files</a></li>
        <li><a href="#span-study-cases">Study Cases</a></li>
        <li><a href="#span-faq">FAQ</a></li>
    </ul>
</div>

<h2 id="span-features">Features</h2>
<ul>
    <li>Part of integrated peak calling <a href="tools.html">solution</a></li>
    <li>Works with both conventional and ultra-low-input ChIP-seq data</li>
    <li>Works with both narrow and wide modifications</li>
    <li>Capable to process tracks with different signal-to-noise ratio</li>
    <li>Supports optional control track</li>
    <li>Supports replicates on model level</li>
    <li>False Discovery Rate correction</li>
    <li>Experimental: differential peak calling</li>
</ul>


<h2 id="span-installation">Installation</h2>
SPAN Peak Analyzer (build @BUILD@), released on @DATE@
<table class="table table-bordered table-nonfluid">
    <thead class="thead-light">
    <tr>
        <th>Download</th>
        <th>Description</th>
    </tr>
    </thead>
    <tbody>
    @TABLE@
    </tbody>
</table>
Requirements:
<ol>
    <li>Download and install <a href="http://www.java.com/en/download/" rel="nofollow">Java 8</a>.</li>
    <li>Download the <code>&lt;build&gt;.chrom.sizes</code> chromosome sizes of the organism you want to analyze
        from the UCSC <a href="http://hgdownload.cse.ucsc.edu/downloads.html" rel="nofollow">website</a>. <br>
        Here is <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes">the file</a> used in
        our
        study.
    </li>
</ol>
<h2 id="span-usage">Usage of SPAN</h2>
<pre><code>java -Xmx1G -jar span.jar [-h] [--version] analyze</code></pre>
<p>
    Use <code>java</code> <code class="code-grey">-Xmx</code> memory settings to limit memory usage. 1 gigabyte
    is used in examples.
</p>
<table class="table table-bordered table-nonfluid">
    <tbody>
    <tr>
        <th>Example of regular peak calling</th>
        <td class="code-grey"><code style="color:black">java -Xmx1G -jar span.jar analyze -t ChIP.bam -c Control.bam
            --cs
            Chrom.sizes \<br>&nbsp;&nbsp;&nbsp;-o Peaks.bed</code></td>
    </tr>
    <tr>
        <th>Example of supervised peak calling</th>
        <td class="code-grey"><code style="color:black">java -Xmx1G -jar span.jar analyze -t ChIP.bam -c Control.bam
            --cs
            Chrom.sizes \<br>&nbsp;&nbsp;&nbsp;-l Labels.bed -o Peaks.bed</code></td>
    </tr>
    <tr>
        <th>Example of model fitting</th>
        <td class="code-grey"><code style="color:black">java -Xmx1G -jar span.jar analyze -t ChIP.bam -c Control.bam
            --cs
            Chrom.sizes</code></td>
    </tr>
    </tbody>
</table>
<h3 id="span-peak-calling">Peak calling</h3>
<p>To analyze a single (possibly replicated) biological condition use <code>analyze</code> command.</p>
<p>
<strong>-b, --bin BIN_SIZE</strong><br>
Peak analysis is performed on read coverage tiled into consequent bins, with size being configurable. Default value
is 200bp, approximately the length of one nucleosome.
</p>
<p>
    <strong>-t, --treatment TREATMENT</strong><br>
    <strong>Required</strong>. ChIP-seq treatment file. Supported formats: BAM, BED, or BED.gz file. If multiple files
    are given, treated as replicates.
    Multiple files should be separated by commas: <code class="code-grey">-t
    A,B,C</code>. Multiple files are processed as replicates on model level.
</p>
<p>
    <strong>-c, --control CONTROL</strong><br>
    Control file. Multiple files should be separated by commas. Single control file or separate file per each treatment
    file required.
    <br>
    Follow instructions for <code class="code-grey">-t, --treatment TREATMENT</code>.
</p>
<p>
<strong>-cs, --chrom.sizes CHROMOSOMES_SIZES</strong><br>
<strong>Required</strong>. Chromosome sizes file for genome build used in <code class="code-grey">TREATMENT</code> and
<code class="code-grey">CONTROL</code> files.
<br>
Can be downloaded at <code class="code-grey">http://hgdownload.cse.ucsc.edu/goldenPath/&lt;build&gt;/bigZips/&lt;build&gt;.chrom.sizes</code>
</p>
<p>
<strong>--fragment FRAGMENT</strong><br>
Fragment size used to convert reads into tags. Default value is half the read length.
</p>
<p>
<strong>-o, --output OUTPUT</strong><br>
Resulting BED file. If omitted, only model fitting step is performed.
</p>
<p>
<strong>-f, --fdr FDR</strong><br>
Minimum FDR cutoff to call significant regions, default value is <code>1.0E-6</code>.
<br>
<code>SPAN</code> reports p- and q- values for the
null hypothesis that a given bin is not enriched with a histone modification. Peaks are formed from a list of truly (in the FDR sense)
enriched bins for the analyzed biological condition by thresholding the Q-value with a cutoff <code>FDR</code>
and merging spatially close peaks using <code>GAP</code> option to broad ones. This is equivalent to controlling FDR.
<br>q-values are are calculated from p-values using Benjamini-Hochberg procedure.
</p>
<p>
<strong>-g, --gap GAP</strong><br>
Gap size to merge spatially close peaks. Useful for wide histone modifications. Default value is 5, i.e. peaks separated
by 5*<code class="code-grey">BIN</code> distance or less are merged.
</p>
<p>
<strong>--labels LABELS</strong><br>
Labels BED file. Used in <a href="tools.html">semi-supervised peak calling</a>.
</p>
<p>
<strong>-d, --debug</strong><br>
Print all the debug information, used for troubleshooting.
</p>
<p>
<strong>-q, --quiet</strong><br>
Turn off output.
</p>
<p>
<strong>-w, --workdir PATH</strong><br>
Path to the working directory (stores coverage and model caches).
</p>
<p>
<strong>--threads THREADS</strong><br>
Configures parallelism level.
</p>
<h3 id="span-supervised-peak-calling">Supervised peak calling</h3>
When <code class="code-grey">LABELS</code> parameter is given, it is used to optimize peak caller parameters for markup.
<br>
<br>
<h3 id="span-model-fitting">Model fitting</h3>
<code>SPAN</code> workflow consists of several steps:
<ol>
    <li>Convert raw reads to tags using <code class="code-grey">FRAGMENT</code> parameter.</li>
    <li>Compute coverage for all genome tiled into bins of <code class="code-grey">BIN</code> base pairs.</li>
    <li>Fit 3-state hidden Markov model that classifies bins as <code>ZERO</code>
        states with no coverage, <code>LOW</code> states of non-specific binding,
        and <code>HIGH</code> states of the specific binding.
    </li>
    <li>Compute posterior <code>HIGH</code> state probability of each bin.</li>
    <li>Trained model is saved into <code class="code-grey">.tar</code> format.</li>
    <li>Peaks are computed using trained model and <code class="code-grey">FDR</code> and <code
            class="code-grey">GAP</code> parameters.
    </li>
    <li>If <code class="code-grey">LABELS</code> are provided, optimal parameters are computed to conform with them.</li>
</ol>
Model fitting mode produces trained <strong>model file</strong> in binary format as output, which can be:
<ol>
    <li>visualized directly in <a href="jbr.html">JBR Genome Browser</a>
    <li>used in integrated peak calling <a href="tools.html">pipeline</a></li>
</ol>
<br>

<h2 id="span-output-files">Output files</h2>
<ul>
    <li> If <code class="code-grey">OUTPUT</code> file is given, it will contain predicted and FDR-controlled peaks in
        the <a
                href="https://github.com/taoliu/MACS">MACS2</a> format:</p>
        <pre><code>&lt;chromosome&gt; &lt;peak start&gt; &lt;peak end&gt; &lt;peak name&gt; &lt;score&gt; . &lt;coverage / foldchange&gt; &lt;-log p-value&gt; &lt;-log Q-value&gt;
</code></pre>
        <blockquote>
            <ul>
                <li>chromosome name</li>
                <li>start position of peak</li>
                <li>end position of peak</li>
                <li>peak name</li>
                <li>score of the peak, computed as log10(qvalue) * log(peak length). Useful for peak ranking with wide
                    histone modifications.
                </li>
                <li>. (represents strand)</li>
                <li>summary reads coverage in peak averaged over replicates. fold-change in differential mode.</li>
                <li>-log10(pvalue) of null-hypothesis that given peak is in <code>ZERO</code> or <code>LOW</code> state.
                </li>
                <li>-log10(qvalue), calculated from p-values using Benjamini-Hochberg procedure. Median value for merged
                    peak.
                </li>
            </ul>
        </blockquote>
    </li>
    <li>In case of <code>SPAN</code> model fitting, it produces model file in binary format.
        <br> NOTE: after model is trained once, it will be reused automatically in other modes.
    </li>
</ul>

<h2 id="span-study-cases">Study Cases</h2>
<p>As a benchmark we applied SPAN peak calling approach to public conventional ChIP-seq datasets as well as to a ULI
    ChIP-seq dataset.</p>

<p>
    CD14+ classical monocytes tracks available in ENCODE database were a natural choice for a conventional ChIP-seq
    dataset.<br/>
    We also used the data from
    <a href="https://academic.oup.com/bioinformatics/article/33/4/491/2608653">Hocking <i>et al.</i></a>
    to evaluate <code>SPAN</code>.
</p>
<p>
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/25607992">Chen C  <i>et al.</i></a> presented an ultra-low-input
    micrococcal nuclease-based native ChIP (ULI-NChIP) and sequencing method to generate genome-wide histone mark
    profiles
    with high resolution and reproducibility from as few as one thousand cells.
    We used these tracks to estimate semi-supervised approach in extreme conditions.
</p>
<p>
    <code>SPAN</code> produced high quality peak calling in all of these cases, see <a
        href="study_cases.html">report</a>.
    <br>
    This suggests that SPAN Peak Analyzer can be used as a general purpose peak calling solution.
</p>
<br>
<h2 id="span-faq">FAQ</h2>
<p>
    Q: What is average running time?
    <br>
    A: <code>SPAN</code> is capable of processing single ChIP-Seq track in less than 1 hour on moderate laptop (MacBook
    Pro 2015).
</p>
<p>Q: Which operating systems are supported?
    <br>
    A: <code>SPAN</code> is developed in modern <a href="https://kotlinlang.org/?">Kotlin</a> programming language and
    can be executed on any platform supported by <code>java</code>.
</p>
<p>
    Q: Is differential peak calling supported?
    <br>
    A: This is experimental feature, see for details:
    <code class="code-grey">java -Xmx1G -jar span.jar compare -h</code>
</p>
<p>Q: Where is <code>SPAN</code> source code?
    <br>
    A: We are working on open sourcing the code. At the moment only the JAR is available.
</p>

<p>Q: Where did you get this lovely span picture?
    <br>
    A: From <a href="http://ascii.co.uk/art/bridges" rel="nofollow">ascii.co.uk</a>, it seems the original author goes
    by the name <code>jgs</code>.
</p>

